### PR TITLE
feat(ojoi): Added comment screen to application

### PR DESCRIPTION
**Pull request by jonbjarnio**
## What

Added comment screen to retry draft state

## Why

To enable user to communicate with reviewers.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced a new form template for the Official Journal of Iceland application, enabling users to draft and retry submissions with detailed sections and a submission button.
	- Added a new message definition for sending applications in the application's messages section.

- **Enhancements**
	- Improved application states and events handling by adding a new state and refining event types for clearer functionality.
	- Enhanced user experience by updating form imports and adjusting state transitions for submission actions.

- **Refactor**
	- Restructured and renamed enums for better code organization and readability.

### CODE REVIEW

File: libs/application/templates/official-journal-of-iceland/src/forms/DraftRetry.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      ],
+      // condition: (answers: Answers) => {
+      //   const hasComments =
+      //     answers.comments?.comments?.length !== undefined &&
+      //     answers.comments?.comments.length > 0
+      //   if (hasComments) {
+      //     return true
+      //   }
+      //   return false
+      // },
```
Comment (thorkellmani) [inline libs/application/templates/official-journal-of-iceland/src/forms/DraftRetry.ts:51]: Remove?

File: libs/application/templates/official-journal-of-iceland/src/lib/OJOIApplication.ts [hunk 2]
```diff
@@ -119,8 +117,54 @@ const OJOITemplate: ApplicationTemplate<
...
               formLoader: () =>
-                import('../forms/Draft').then((val) =>
-                  Promise.resolve(val.Draft),
+                import('../forms/DraftRetry').then((val) =>
+                  Promise.resolve(val.DraftRetry),
+                ),
+              actions: [
+                {
+                  event: DefaultEvents.SUBMIT,
+                  name: 'Senda umsókn',
```
Comment (thorkellmani) [inline libs/application/templates/official-journal-of-iceland/src/lib/OJOIApplication.ts:126]: Won't this be translated?

**Review by thorkellmani — APPROVED** [review]
LGTM

File: libs/application/templates/official-journal-of-iceland/src/forms/DraftRetry.ts (hunk 1 above)
Comment (jonbjarnio) [reply libs/application/templates/official-journal-of-iceland/src/forms/DraftRetry.ts:51]: Þetta á við í framtíðinni, get tekið út á meðan.

**Pull request merged**